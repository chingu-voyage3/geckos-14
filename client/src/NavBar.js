import React, { Component } from 'react';
import {NavLink} from 'react-router-dom'
class NavBar extends Component {
  render() {
    return (
      <div>
      <ul>
        <li>DashThing Logo</li>
        <li><NavLink to="/">Home</NavLink></li>
        <li><NavLink to="/getstarted">Get Started</NavLink></li>
        <li><NavLink to="/dashboard">Dashboard</NavLink></li>
      </ul>
    </div>
    );
  }
}

export default NavBar;
