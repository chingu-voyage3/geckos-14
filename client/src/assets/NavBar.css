.nav-bar {
  position: sticky;
  top: 0;
}

.nav-bar a {
}

ul {
  padding: 0;
  list-style-type: none;
}

.nav-bar ul {
  margin-left: 0;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(4, 15%);
  -ms-grid-column: repeat(4, 15%);
}
