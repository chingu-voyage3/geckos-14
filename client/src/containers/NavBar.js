import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../assets/NavBar.css';

class NavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLogged: false
    };
  }
  handleLogin = () => {
    if (!this.state.isLogged) {
      this.setState({ isLogged: true });
    }
  };
  handleLogout = () => {
    if (this.state.isLogged) {
      this.setState({ isLogged: false });
    }
  };
  renderLogin() {
    if (this.state.isLogged) {
      return (
        <ul>
          <li>
            <Link to="/profile">Profile</Link>
          </li>
          <li>
            <button onClick={this.handleLogout}>LogOut</button>
          </li>
        </ul>
      );
    }

    return (
      <ul>
        <li>
          <button onClick={this.handleLogin}>LogIn</button>
        </li>
        <li>
          <Link to="/register">Register</Link>
        </li>
      </ul>
    );
  }
  render() {
    return (
      <div className="nav-bar">
        <ul>
          <li>DashThing Logo</li>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/getstarted">Get Started</Link>
          </li>
          <li>
            <Link to="/dashboard">Dashboard</Link>
          </li>
        </ul>
        {this.renderLogin()}
      </div>
    );
  }
}

export default NavBar;
