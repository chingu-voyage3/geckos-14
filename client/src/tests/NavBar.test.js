import React from 'react';
import NavBar from '../containers/NavBar';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const wrapper = shallow(<NavBar />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
