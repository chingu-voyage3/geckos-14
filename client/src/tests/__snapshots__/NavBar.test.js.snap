// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="nav-bar"
>
  <ul>
    <li>
      DashThing Logo
    </li>
    <li>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/getstarted"
      >
        Get Started
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/dashboard"
      >
        Dashboard
      </Link>
    </li>
  </ul>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        LogIn
      </button>
    </li>
    <li>
      <Link
        replace={false}
        to="/register"
      >
        Register
      </Link>
    </li>
  </ul>
</div>
`;
